src/main/java/seedu/address/model/person/Person.java

```diff
@@ -52,6 +71,18 @@ public Email getEmail() {
         return email;
     }
 
+    public Favourite getFavourite() {
+        return favourite;
+    }
+
+    /**
+     * Toggles the favourite status of Person
+     */
+    public void toggleFavourite() {
+        boolean toggledStatus = !favourite.getStatus();
+        favourite.setStatus(toggledStatus);
+    }
```

Comment:
Will this violate the guarantee that field values are immutable?

Reply:
I'm not sure if I understand this correctly, but if you mean does it change the value of `Favourite` field of a Person, then yes this field should be mutable since the user is able to 'Favourite' or 'Unfavourite' a Person based on his/her `Favourite `status currently.

Reply:
The reason im asking is because the javadoc for the class currently states that all fields are immutable.

So we need to either remove this guarantee or make this method return a new instance of `Person`

Reply:
I think it would be more logical to change the field of the `Person` rather than return a new instance of `Person`, what do you think?

Reply:
Im fine with that but make sure to update the javadoc